paddle/fluid/framework/ir/graph_pattern_detector.cc

```diff
@@ -958,6 +958,24 @@ PDNode *patterns::OperatorActivation::operator()(
   return activation_out;
 }
 
+PDNode *patterns::Squeeze2Transpose2::operator()() {
+  auto *preceding_op_out = pattern->NewNode(preceding_op_out_repr())
+                               ->AsInput()
+                               ->assert_is_op_input("squeeze2", "X");
+  auto *squeeze2_op =
+      pattern->NewNode(squeeze2_op_repr())->assert_is_op("squeeze2");
+  auto *squeeze2_op_out = pattern->NewNode(squeeze2_op_out_repr())
+                              ->AsIntermediate()
```

Comment:
`assert_has_n_outputs(2)` (`Out` and `XShape`) is missing here. We cannot fuse squeeze2+transpose2 if transpose2 is not the only output of squeeze2